pacai/util/probability.py

```diff
@@ -4,27 +4,23 @@
 
 import random
 
-from pacai.util import counter
-
-def normalize(vectorOrCounter):
+def normalize(vectorOrDict):
     """
-    Normalize a vector or counter by dividing each value by the sum of all values.
+    Normalize a vector or dictionary by dividing each value by the sum of all values.
     """
 
-    normalizedCounter = counter.Counter()
-    if type(vectorOrCounter) == type(normalizedCounter):
-        counterContainer = vectorOrCounter
-        total = float(counterContainer.totalCount())
+    if type(vectorOrDict) == dict:
+        total = float(sum(vectorOrDict.values()))
         if total == 0:
-            return counterContainer
+            return vectorOrDict
 
-        for key in list(counter.keys()):
-            value = counter[key]
-            normalizedCounter[key] = value / total
+        normalizedDict = {}
+        for key, value in vectorOrDict.items():
+            normalizedDict[key] = value / total
 
-        return normalizedCounter
+        return normalizedDict
     else:
-        vector = vectorOrCounter
+        vector = vectorOrDict
         s = float(sum(vector))
```

Comment:
Always avoid single letter variable names.
The dict part uses 'total', try to keep it consistent.